### ISSUE TITLE
MPC feature 'gendir' doesn't work with OpenDDS

### ISSUE THREAD

**Issue by jwillemsen** [objectcomputing/OpenDDS#423]
We normally put all generated files into a subdirectory to keep our user code directory clean. As a test I changed the ReadCondition test to have:
```
project: dcpsexe, dcps_transports_for_test {
  exename = ReadConditionTest
  idlflags += -SS
  TypeSupport_Files {
    gendir = GeneratedCode
    Messenger.idl
  }
}
```
This gives the command below, see that `-o` isn't passed to opendds_idl to put all generated files into a subdirectory

```
mkdir -p GeneratedCode
OpenDDS/bin/opendds_idl -Sa -St Messenger.idl
```

**Comment by jwillemsen** [comment]
Should have added 
```
  idlflags += -SS -o GeneratedCode
  dcps_ts_flags += -o GeneratedCode
```
But that now gives other errors

**Comment by jwillemsen** [comment]
@mitza-oci Do you have an idea where to look in OpenDDS for this?

**Comment by jwillemsen** [comment]
Doing a `git clean -fxd .` and regenerate the MPC file solved the warnings but it seems we compile the files now four times

```
processing Messenger.idl
/ACE/ACE/bin/tao_idl -Wb,pre_include=ace/pre.h -Wb,post_include=ace/post.h -I/ACE/TAO -Sa -St -I../../.. -SS -o GeneratedCode GeneratedCode/MessengerTypeSupport.idl
processing GeneratedCode/MessengerTypeSupport.idl
/ACE/ACE/bin/tao_idl -Wb,pre_include=ace/pre.h -Wb,post_include=ace/post.h -I/ACE/TAO -Sa -St -I../../.. -SS -o GeneratedCode GeneratedCode/MessengerTypeSupport.idl
processing GeneratedCode/MessengerTypeSupport.idl
/ACE/ACE/bin/tao_idl -Wb,pre_include=ace/pre.h -Wb,post_include=ace/post.h -I/ACE/TAO -Sa -St -I../../.. -SS -o GeneratedCode GeneratedCode/MessengerTypeSupport.idl
processing GeneratedCode/MessengerTypeSupport.idl
/ACE/ACE/bin/tao_idl -Wb,pre_include=ace/pre.h -Wb,post_include=ace/post.h -I/ACE/TAO -Sa -St -I../../.. -SS -o GeneratedCode GeneratedCode/MessengerTypeSupport.idl
processing GeneratedCode/MessengerTypeSupport.idl
/ACE/ACE/bin/tao_idl -Wb,pre_include=ace/pre.h -Wb,post_include=ace/post.h -I/ACE/TAO -Sa -St -I../../.. -SS -o GeneratedCode Messenger.idl
processing Messenger.idl
/ACE/ACE/bin/tao_idl -Wb,pre_include=ace/pre.h -Wb,post_include=ace/post.h -I/ACE/TAO -Sa -St -I../../.. -SS -o GeneratedCode Messenger.idl
processing Messenger.idl
/ACE/ACE/bin/tao_idl -Wb,pre_include=ace/pre.h -Wb,post_include=ace/post.h -I/ACE/TAO -Sa -St -I../../.. -SS -o GeneratedCode Messenger.idl
processing Messenger.idl
/ACE/ACE/bin/tao_idl -Wb,pre_include=ace/pre.h -Wb,post_include=ace/post.h -I/ACE/TAO -Sa -St -I../../.. -SS -o GeneratedCode Messenger.idl
processing Messenger.idl
```

**Comment by jwillemsen** [comment]
Looks a few things are broken in the generated GNUmakefile.ReadCondition
```
GENERATED_DIRTY += GeneratedCode/GeneratedCode/MessengerTypeSupport.idl GeneratedCode/MessengerTypeSupportImpl.cpp GeneratedCode/MessengerTypeSupportImpl.h
GENERATED_DIRTY += MessengerC.cpp MessengerS.h MessengerC.h MessengerC.inl
```

**Comment by jwillemsen** [comment]
Made some changes as part of issue #427, this can be done, see ReadCondition, propose to merge that as example

**Comment by mitza-oci** [comment]
See pull request #458, this is relatively clean.  It could be cleaner depending on what changes we're willing to make in the MPC and ACE_TAO repositories.  For this to work on -type vc10 through vc14 the latest MPC from GitHub master is required.

**Comment by jwillemsen** [comment]
Will update my PR, personally I would like to make this clean, even when it results in some ace and mpc changes

**Comment by jwillemsen** [comment]
Closed, fixed through issue #427 and a fix in MPC
